qiskit/quantum_info/operators/unitary.py

```diff
@@ -0,0 +1,250 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+# pylint: disable=invalid-name,assignment-from-no-return
+
+"""
+Tools for working with Unitary Operators.
+
+A simple unitary class and some tools.
+"""
+import copy
+import numpy
+import sympy
+from qiskit.exceptions import QiskitError
+from qiskit.circuit.gate import Gate
+from qiskit.dagcircuit import DAGCircuit
+
+
+class Unitary(Gate):
+    """Class for representing unitary operators"""
+
+    def __init__(self, matrix, *qargs, label=None, validate=True, rtol=1e-5, atol=1e-8):
+        """
+        Create unitary.
+
+        Args:
+            matrix (numpy.ndarray or list(list)): unitary matrix representation.
+            qargs (list(tuple(QuantumCircuit, int))): qubits to apply operation to.
+            label (str): identifier hint for backend
+            validate (bool): whether to validate unitarity of matrix when supplied
+                either here or with Unitary.matrix.
+            rtol (float): relative tolerance (see numpy.allclose).
+            atol (float): absolute tolerance (see numpy.allclose).
+        """
+        self._matrix = None
+        self._label = None
```

Comment:
is this line needed? a few lines down, the label is set again from the arg.

Reply:
Without these we get the pylint error `attribute-defined-outside-init`.